pkg/genericapiserver/endpoints/handlers/rest.go

```diff
@@ -140,7 +140,7 @@ func GetResource(r rest.Getter, e rest.Exporter, scope RequestScope) restful.Rou
 			options := metav1.GetOptions{}
 			if values := req.Request.URL.Query(); len(values) > 0 {
 				exports := metav1.ExportOptions{}
-				if err := scope.ParameterCodec.DecodeParameters(values, schema.GroupVersion{Version: "v1"}, &exports); err != nil {
+				if err := metainternalversion.ParameterCodec.DecodeParameters(values, scope.MetaGroupVersion, &exports); err != nil {
```

Comment:
The type alias doesn't matter - internal type isn't really a thing.  It's better to think of "I'm asking for a particular version of this struct", where someone else is responsible for hiding the details of giving you that version.  An external version is not special.  Creating a type alias is just a second form of alias that doesn't actually clear things up.  

The contract is "I take request parameters and give you back an object".  Internal version should look "special" or have a deeper reason to exist.  Just type aliasing it won't reduce that.